### PR TITLE
[pytorch][ao] Add `torch.matmul` in FloatFunctional/QFunctional

### PR DESCRIPTION
**Pull request by jiaxuzhu92**
Summary: As title

Test Plan: new unit tests

Differential Revision: D48172841

### CODE REVIEW

File: torch/ao/nn/quantized/modules/functional_modules.py [hunk 1]
```diff
@@ -79,6 +79,12 @@ def add_relu(self, x: Tensor, y: Tensor) -> Tensor:
         r = self.activation_post_process(r)
         return r
 
+    r"""Operation equivalent to ``torch.matmul(Tensor, Tensor)``"""
+    def matmul(self, x: Tensor, y: Tensor) -> Tensor:
```
Comment (jerryzh168) [inline torch/ao/nn/quantized/modules/functional_modules.py:83]: can you make corresponding changes in `FXFloatFunctional` as well?

File: torch/ao/nn/quantized/modules/functional_modules.py [hunk 2]
```diff
@@ -220,6 +226,12 @@ def add_relu(self, x: Tensor, y: Tensor) -> Tensor:
         r = self.activation_post_process(r)
         return r
 
+    r"""Operation equivalent to ``torch.ops.quantized.matmul(Tensor, Tensor)``"""
+    def matmul(self, x: Tensor, y: Tensor) -> Tensor:
+        r = ops.quantized.matmul(x, y, scale=self.scale, zero_point=self.zero_point)
+        r = self.activation_post_process(r)
```
Comment (jerryzh168) [inline torch/ao/nn/quantized/modules/functional_modules.py:232]: I feel we probably don't need to call this, could you remove this line and test again to see if it still works?

Comment (jiaxuzhu92) [reply torch/ao/nn/quantized/modules/functional_modules.py:232]: Can you elaborate why we don't need `matmul` in QFunctional? the eager mode convert will swap FloatFunctional to QFunctional right?

Comment (jerryzh168) [reply torch/ao/nn/quantized/modules/functional_modules.py:232]: no I mean we don't need to call `        r = self.activation_post_process(r)`

File: torch/ao/nn/quantized/modules/functional_modules.py [hunk 3]
```diff
@@ -220,6 +231,13 @@ def add_relu(self, x: Tensor, y: Tensor) -> Tensor:
         r = self.activation_post_process(r)
         return r
 
+    r"""Operation equivalent to ``torch.ops.quantized.matmul(Tensor, Tensor)``"""
+    def matmul(self, x: Tensor, y: Tensor) -> Tensor:
+        r = ops.quantized.matmul(x, y, scale=self.scale, zero_point=self.zero_point)
+        # Note: this operation is not observed because the observation is not
+        # needed for the quantized op.
```
Comment (jerryzh168) [inline torch/ao/nn/quantized/modules/functional_modules.py:238]: you can also remove this comment I think, it is observed, but we don't need to call `activation_post_process` in quantized model

**Review by jerryzh168 — APPROVED** [review]
lg, thanks! please see one comment inline

**Pull request closed without merging**